Adds batch 4 of Bug doc text to RNs (#48326)
openshift/openshift-docs: release_notes/ocp-4-11-release-notes.adoc

[id="ocp-4-11-machine-config-operator-bug-fixes"]
==== Machine Config Operator

* Because `keyFile` is not configured as the default plug-in for NetworkManager on RHEL nodes, RHEL nodes might not reach the ready state after a reboot.  With this fix, `keyFile is set as the default NetworkManager plug-in on all cluster nodes. As a result, nodes correctly reach the ready state after a reboot. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2060133[*BZ#2060133*])

> * Because vSphere UPI clusters do not set the `PlatformStatus.VSphere` parameter at installation, the parameter was set to 'nil`. This caused the MCO logs to be filled with unnecessary and repetitive messages that this parameter cannot be `nil`. This fix removes the warning, which was added to resolve a separate issue. As a result, the logs no longer list this message for vSphere U{I installations. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2080267[*BZ#2080267*])

Reviewer: ```suggestion
* Because vSphere UPI clusters do not set the `PlatformStatus.VSphere` parameter at installation, the parameter was set to `nil`. This caused the MCO logs to be filled with unnecessary and repetitive messages that this parameter cannot have the value `nil`. This fix removes the warning, which was added to resolve a separate issue. As a result, the logs no longer list this message for vSphere UPI installations. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2080267[*BZ#2080267*])
```

[id="ocp-4-11-node-bug-fixes"]
==== Node

> * Previously, the pod manager handled the registration and de-registration  of pod secrets and config maps. Because of this, pod secrets would sometimes fail to be mounted within a pod. With this fix, the pod id is included in the key that kublet uses to manage registered pods. As a result, secrets are proper mounted as expected. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1999325[*BZ#1999325*])

Reviewer: ```suggestion
* Previously, the pod manager handled the registration and de-registration of pod secrets and config maps. Because of this, pod secrets would sometimes fail to be mounted within a pod. With this fix, the pod ID is included in the key that the kubelet uses to manage registered pods. As a result, secrets are properly mounted as expected. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1999325[*BZ#1999325*])
```

> * Because of a memory leak in the garbage collection process, pods might not be able to start on a node due to lack of memory. With this fix  memory is no longer leaking in the garbage collection prodess and nodes should start as expected. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2065749[*BZ#2065749*])

Reviewer: ```suggestion
* Because of a memory leak in the garbage collection process, pods might not be able to start on a node due to lack of memory. With this fix memory is no longer leaking in the garbage collection process and nodes should start as expected. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2065749[*BZ#2065749*])
```

* Because of a change in the upstream Kubernetes, the kubelet was not running readiness probes on terminated pods. As a result, a load balancer or controller could react more slowly to a terminating pod, which might have resulted in errors. With this fix, readiness probes are again being performed on pod termination. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2089933[*BZ#2089933*])

> * Because of a bug, kubelet could incorrectly reject pods that have `OutOfCpu` errors, if the pods were rapidly scheduled after other pods were reported as complete in the API. With this fix, the Kubelet now waits to report the phase of a pod as terminal in the API until all running containers have stopped and no new containers have been started. Short-lived pods may take slightly longer (~1s) to report success or failure after this change. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2022507[*BZ#2022507*])

Reviewer: ```suggestion
* Because of a bug, kubelet could incorrectly reject pods that have `OutOfCpu` errors, if the pods were rapidly scheduled after other pods were reported as complete in the API. With this fix, the Kubelet now waits to report the phase of a pod as terminal in the API until all running containers have stopped and no new containers have been started. Short-lived pods may take slightly longer, approximately 1s, to report either success or failure after this change. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2022507[*BZ#2022507*])
```
Also, I'm not sure about what article to use for the first instance of "kubelet", it seems like either "a" or "the" would work, but I don't know what is standard. The second, capitalized instance of "Kubelet" also caught my eye since it looked strange but I see instances of both lowercase and uppercase across the Kubernetes docs.

Reviewer 2: I try to use _the kubelet_.

> * Because of a bug, kubelet could incorrectly reject pods that have `OutOfCpu` errors, if the pods were rapidly scheduled after other pods were reported as complete in the API. With this fix, the Kubelet now waits to report the phase of a pod as terminal in the API until all running containers have stopped and no new containers have been started. Short-lived pods may take slightly longer (~1s) to report success or failure after this change. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2022507[*BZ#2022507*])

Reviewer 2: s/the Kubelet/the kubelet

> * Because recent versions of the `prometheus-adapter` are sending additional pod metrics, the Vertical Pod Autoscaler (VPA) recommender is producing a large number of unnecessary and repetitive messages. With this fix,  the VPA recognizes and ignores the extra metrics. As a result, these messages are no longer being produced. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2102947[*BZ#2102947*])

Reviewer: ```suggestion
* Because recent versions of the `prometheus-adapter` are sending additional pod metrics, the Vertical Pod Autoscaler (VPA) recommender is producing a large number of unnecessary and repetitive messages. With this fix, the VPA recognizes and ignores the extra metrics. As a result, these messages are no longer being produced. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2102947[*BZ#2102947*])
```
